### PR TITLE
Add info about ownership management and SELinux relabeling to volume sources

### PR DESCRIPTION
**Pull request by pmorie**
@thockin @swagiaal @kubernetes/rh-storage

### CODE REVIEW

File: api/swagger-spec/v1.json [hunk 1]
```diff
@@ -12733,7 +12733,7 @@
    },
    "v1.AWSElasticBlockStoreVolumeSource": {
     "id": "v1.AWSElasticBlockStoreVolumeSource",
-    "description": "Represents a persistent disk resource in AWS.\n\nAn Amazon Elastic Block Store (EBS) must already be created, formatted, and reside in the same AWS zone as the kubelet before it can be mounted. Note: Amazon EBS volumes can be mounted to only one instance at a time.",
+    "description": "AWSElasticBlockStoreVolumeSource represents a Persistent Disk resource in AWS.\n\nAn AWS EBS disk must exist and be formatted before mounting to a container. The disk must also be in the same AWS zone as the kubelet. A AWS EBS disk can only be mounted as read/write once. AWS EBS volumes support ownership management and SELinux relabeling.",
```
Comment (swagiaal) [inline api/swagger-spec/v1.json]: s/A AWS/An AWS/

File: api/swagger-spec/v1.json [hunk 2]
```diff
@@ -12909,7 +12909,7 @@
    },
    "v1.CinderVolumeSource": {
     "id": "v1.CinderVolumeSource",
-    "description": "CinderVolumeSource represents a cinder volume resource in Openstack. A Cinder volume must exist before mounting to a container. The volume must also be in the same region as the kubelet.",
+    "description": "CinderVolumeSource represents a cinder volume resource in Openstack. A Cinder volume must exist and be formatted before mounting to a container. The volume must also be in the same region as the kubelet. Cinder volumes support ownership management and SELinux relabeling.",
```
Comment (swagiaal) [inline api/swagger-spec/v1.json]: Cinder volumes do not have to be formatted. The cinder volume implements its own version of FormatAndMount. Incidentally that should use mount.FormatAndMount

File: api/swagger-spec/v1.json [hunk 3]
```diff
@@ -13235,7 +13235,7 @@
    },
    "v1.EmptyDirVolumeSource": {
     "id": "v1.EmptyDirVolumeSource",
-    "description": "EmptyDirVolumeSource is temporary directory that shares a pod's lifetime.",
+    "description": "EmptyDirVolumeSource represents an empty directory for a pod. Empty directory volumes support ownership management and SELinux relabeling.",
```
Comment (swagiaal) [inline api/swagger-spec/v1.json]: Should we leave "temporary" there ? EmptyDir is meant to be temporary storage no ?

File: pkg/api/types.go [hunk 4]
```diff
@@ -386,12 +386,14 @@ const (
 	ClaimBound PersistentVolumeClaimPhase = "Bound"
 )
 
-// HostPathVolumeSource represents a host directory mapped into a pod.
+// HostPathVolumeSource represents a host path mapped into a pod.
+// Host path volumes do not support ownership management or SELinux relabeling.
 type HostPathVolumeSource struct {
```
Comment (swagiaal) [inline pkg/api/types.go]: I think the struct name was used here to conform with godoc. Is this not a requirement ?

File: pkg/api/types.go [hunk 5]
```diff
@@ -386,12 +386,14 @@ const (
 	ClaimBound PersistentVolumeClaimPhase = "Bound"
 )
 
-// HostPathVolumeSource represents a host directory mapped into a pod.
```
Comment (swagiaal) [inline pkg/api/types.go]: I think the struct name was used here to conform with godoc. Is this not a requirement ?

File: pkg/api/types.go [hunk 6]
```diff
@@ -386,12 +386,14 @@ const (
...
+// HostPathVolumeSource represents a host path mapped into a pod.
```
Comment (swagiaal) [inline pkg/api/types.go]: Ah my bad :)

**Comment by swagiaal** [comment]
just a few nits. Otherwise LGTM.

File: api/swagger-spec/v1.json (hunk 3 above)
Comment (pmorie) [inline api/swagger-spec/v1.json]: I'll add another note about that in.  'Temporary' isn't quite right -- it's scoped to the lifetime of the pod.  There's been some confusion about this in the community and we likely need to improve the docs too.

File: api/swagger-spec/v1.json (hunk 2 above)
Comment (pmorie) [inline api/swagger-spec/v1.json]: Can you make a suggestion for a better comment here?

File: api/swagger-spec/v1.json (hunk 1 above)
Comment (pmorie) [inline api/swagger-spec/v1.json]: Good catch, will fix.

File: api/swagger-spec/v1.json (hunk 2 above)
Comment (swagiaal) [inline api/swagger-spec/v1.json]: Just s/and be formatted //

**Comment by pmorie** [comment]
@swagiaal PTAL

**Comment by swagiaal** [comment]
@pmorie LGTM

**Comment by pmorie** [comment]
@k8s-bot test this

**Comment by pmorie** [comment]
@bgrant0607 for API docs change review

**Comment by bgrant0607** [comment]
Users should never be exposed to Go type and field names. For fields, use json field names. For types, just use English.

**Comment by swagiaal** [comment]
> Users should never be exposed to Go type and field names. For fields, use json field names. For types, just use English.

Ah good point. I'll note that in future review.

**Comment by pmorie** [comment]
@bgrant0607 can you give an example of where this PR breaks that rule?  The docs on the types are similar to what we have for other types (viz: [Pod](http://kubernetes.io/v1.1/docs/api-reference/v1/definitions.html#_v1_pod)).  Perhaps I'm missing something obvious?

**Comment by smarterclayton** [comment]
Basically, don't use the internal Go name in the description.  Use the english words, or use the json value.  If Pod is broken, Pod is also broken.

**Comment by eparis** [comment]
So basically the opposite of go doc/lint requirements.  That's fine, but this PR isn't changing those (except maybe 1 or 2 and is in fact making them consistent...)

**Comment by pmorie** [comment]
I'm happy to change them all, just not in this PR.  I'll make a follow-up that changes them all.

**Comment by smarterclayton** [comment]
Since the godoc is dual use for humans as API doc and Godoc, the decision
was to give precedence to API doc, and humans reading API doc know nothing
about Go.

**Comment by pmorie** [comment]
@eparis @smarterclayton I updated, ptal

**Comment by pmorie** [comment]
Created #18675 to fix the remaining instances in the types.go files.

**Comment by eparis** [comment]
Adding LGTM. At this point the PR is a single new `omitempty` and docs changes. So even though I have a great phobia of types.go for fear of the wrath of @bgrant0607 I'm being bold.

**Pull request merged**